STYLE.md

```diff
@@ -0,0 +1,78 @@
+# Style Guide
+
+The eigenda repo consists majorly of Go and Solidity code, so we break it up into two sections:
+- [Go Style Guide](#go-style-guide)
+- [Solidity Style Guide](#solidity-style-guide)
+
+Our implicit rule is to defer to each language's respective style guides (see [resources](#resources) section). 
+However, some disagreements keep coming up in PRs, so we document our conventions here to prevent too much bike-shedding.
+This style guide is thus by definition perpetually a work in progress.
+
+## Go Style Guide
+
+This document outlines the coding standards and best practices for our Go projects.
+
+### Code Organization
+
+Follow the [Ben Johnson](https://medium.com/@benbjohnson/structuring-applications-in-go-3b04be4ff091) repo structuring method:
+- Write interfaces in a file at the root of each package, and
+- Write every implementation of that interface in its own package in a subdirectory.
+
+#### Function Organization
+- Keep functions small and focused
+- Limit function length to ~50 lines where possible
+- Place the most important functions at the top of the file
+- Public functions that aren't methods should be placed in files with `_utils.go` suffix
```

Comment:
This isn't always the case, i.e. constructors. Perhaps reword to say something like:

"Public static functions that lack a tight coupling to a specific struct should be placed in files with `_utils.go` suffix"

Reply:
```
"Public static functions that lack a tight coupling to a specific struct (e.g. a constructor) should be placed in files with _utils.go suffix"
```